### PR TITLE
Fixing payload for failureCriteria and updated test recordings

### PR DESCRIPTION
**Pull request by Himanshu49**
---
Fix for sending correct failureCriteria request payload when using `az load test create` and `az load test update` commands with `--load-test-config-file` option and config file having failureCriteria.

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/load/azext_load/data_plane/utils/utils.py [hunk 1]
```diff
@@ -445,13 +447,15 @@ def create_or_update_test_with_config(
     new_body["loadTestConfiguration"]["quickStartTest"] = False
 
     new_body["passFailCriteria"] = {}
+    # make every metric in passFailMetrics as None to remove it from the test and append metrics from yaml
     for key in body.get("passFailCriteria", {}):
```
Comment (prativen) [inline src/load/azext_load/data_plane/utils/utils.py:451]: where is key being used?

File: src/load/azext_load/data_plane/utils/utils.py [hunk 2]
```diff
@@ -445,13 +447,15 @@ def create_or_update_test_with_config(
...
-        if "passFailCriteria" not in new_body:
-            new_body["passFailCriteria"] = {}
-        new_body.get("passFailCriteria", {})[key] = None
-        new_body["passFailCriteria"][key] = None
+        new_body["passFailCriteria"]["passFailMetrics"] = {}
```
Comment (prativen) [inline src/load/azext_load/data_plane/utils/utils.py:452]: where is key being used?

File: src/load/azext_load/data_plane/utils/utils.py [hunk 3]
```diff
@@ -324,22 +326,22 @@ def convert_yaml_to_test(data):
...
-            )[
-                1
-            ].strip()
+            new_body["passFailCriteria"]["passFailMetrics"][metric_id]["condition"] = (
+                components.split(")")[1].strip()[0]
+            )
+            new_body["passFailCriteria"]["passFailMetrics"][metric_id]["value"] = (
+                components.split(
+                    new_body["passFailCriteria"]["passFailMetrics"][metric_id][
+                        "condition"
```
Comment (mitsha-microsoft) [inline src/load/azext_load/data_plane/utils/utils.py:341]: nit: move to single line `new_body["passFailCriteria"]["passFailMetrics"][metric_id]["condition"]` ?

File: src/load/azext_load/data_plane/utils/utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Himanshu49) [reply src/load/azext_load/data_plane/utils/utils.py:452]: Yes, that's the behavior expected while updating test using config files. Same behavior is followed in ADO.

Comment (Himanshu49) [reply src/load/azext_load/data_plane/utils/utils.py]: refactored code

File: src/load/azext_load/data_plane/utils/utils.py (hunk 3 above)
Comment (Himanshu49) [reply src/load/azext_load/data_plane/utils/utils.py]: fixed

**Review by necusjz — APPROVED** [review]

**Pull request merged**